1.16.5 to 1.17 update

Dragon-Seeker opened the pull request:
I'm targeting the main 1.16 Dev branch for now but A separate branch should be created for this PR.

This PR contains a set changes to get tropicraft Forge working on 1.17 Minecraft. Currently listed are the changes made the directly impacted the code base:

- Models
  - Due to the new Entity Model format introduced into 1.17, all the models needed to be converted to this new system.
    -  Some models are minorly broken due to bad copying of code meaning some may have incorrect position values, wrong model block size or a second head specifically on the White Lipped Peccary (I really have no clue how this has happened)
  - General entity rendering has been pushed to a subscriber event and so has the new model layers.
    - This did impact issues with getting the scuba gear model's due to how the original code was storing the models so a work around was put into place
  - Speaking of armor, I had to change how it called for a model by adding a create model method within the PlayerHeadPieceModel and the ScubaModel
    - Theses are not working as of the moment as it will show the texture incorrectly or like the ashen mask, it will show a red box around the players head
---------------------------------------------------------------------------------------
- Mapping
  - The mapping has been updated to use mojmap as MCP is no longer support(RIP), but a combination of mojmap and Parchment should be fine replacment
  - Forge has also changed the naming convetion of tile entity's to block entitys so with the future of the PR, such should be reflected on all class files and folders 
---------------------------------------------------------------------------------------
- Dimension
  - Update to how mangrove's work with the removal of methods that were put into the decorator's
  - Change over to Context based methods for a lot of the placers I.E the foliage placer and feature's
  - MAJOR ISSUE: Due to some change between 1.16 to 1.17, access chucks outside the max range of 3x3 for a feature will crash the game so the Home feature should be changed or updated to a structure

Suggestion: I'd like to pr a change that I made within the fabric version of tropicraft the allows for leaf decay on the bigger trees. Its still experimental but would make farming and using tropicraft trees a lot easier. It's not included within this one as Ill do a separate branch within the future.

There might be other stuff missed but for more info on what might needed to be changed: [Link](https://gist.github.com/50ap5ud5/beebcf056cbdd3c922cc8993689428f4)

Commit:
Fix LT logo in Readme

Dragon-Seeker committed:
Mojang Conversion

Outline of the process taken to remap from MCP 1.16 to Mojmap 1.16.5

1. Make sure that the classpath for Forge gradle is this: "classpath group: 'net.minecraftforge.gradle', name: 'ForgeGradle', version: '5.+', changing: true"
   Also you will need to update the gradle version within the gradle-wrapper.properties to: "https\://services.gradle.org/distributions/gradle-7.2-bin.zip"

2.Update the .toImmutableEnumMap map calls within TropicraftItems to include the wildcards so jave knows which method to call

	E.Q. : Maps.<AshenMasks, AshenMasks, RegistryObject<AshenMaskItem>>toImmutableEnumMap(

3.Run this gradlew command within the directory of the tropicraft project: "gradlew -PUPDATE_MAPPINGS=1.16.5 -PUPDATE_MAPPINGS_CHANNEL=official updateMappings"

4.Then update the mapping line to this: "mappings channel: 'official', version: '1.16.5'"

Dragon-Seeker committed:
1.16.5 to 1.17 mapping change

Here is a outline for what I did:

1. Add the code from this website to the build.gradlew: "https://raw.githubusercontent.com/SizableShrimp/Forge-Class-Remapper/main/classremapper.gradle"

2. Run this gradlew command within the directory of the tropicraft project: "gradlew -PUPDATE_CLASSNAMES=true updateClassnames"

3. Then update the mapping line to this: " mappings channel: 'official', version: '1.17.1' "

4. Also change the source compatiblity to: '16' and the forge dependency to latest forge: minecraft 'net.minecraftforge:forge:1.17.1-37.0.13'
	-Would be changed with the gradle.properties like: minecraft_version=1.17.1
						           forge_version=37.0.103

Dragon-Seeker committed:
Change over to include parchment

Outline for what I did:

1. Add parchment maven to repo's within the build.gradle: maven { url = 'https://maven.parchmentmc.org' }

2. Apply the librarian plugin within the build.gradle: apply plugin: 'org.parchmentmc.librarian.forgegradle'

3. Run this command: gradlew -PUPDATE_MAPPINGS=2021.08.29-1.17.1 -PUPDATE_MAPPINGS_CHANNEL=parchment

4. Then change the mappingwithin the build.gradle to this: mappings channel: 'parchment', version: '2021.08.29-1.17.1'

Dragon-Seeker committed:
Bulk of fixing issues within block and item folder

Dragon-Seeker committed:
Porting the models from fabric Repo

Dragon-Seeker committed:
World gen stuff

Dragon-Seeker committed:
Most of the rest world gen ported

Dragon-Seeker committed:
All around fixes and regnerating of assets

Dragon-Seeker committed:
Armor Rendering fix

Dragon-Seeker committed:
Fix floating sunglasses on the tropicraft bee feature

Dragon-Seeker committed:
Fix Two Head issue with White Lipped Peccary

Dragon-Seeker committed:
Fix Basilisk Model issues related to its tail

Dragon-Seeker committed:
Temporarily comment unimplemented tags

Dragon-Seeker committed:
Implement dirty fix for issues with setting blocks outside the loaded region allowed for features

Dragon-Seeker committed:
SuperCoder Addition: Papaya

-Add papaya blocks and generation
-Remove debug, cleanup
-Add some missing features
-Fix merge

Dragon-Seeker committed:
CoderBot: Make golden leather ferns produce way fewer particles when broken

Also fix textures?

Dragon-Seeker committed:
Disable random weight for ocean biomes as it caused a Zero value to be returned if a tropicraft ocean was found

Papaya still cause issues for some reason and cause the world to be only the tropics biome with other issues

Dragon-Seeker committed:
Fix for papaya

Dragon-Seeker committed:
Corosauce: Fix bongo drums pitch adjusting not working in negative coordinates

Dragon-Seeker committed:
Gegy: fix pick-block on jigarbov torches

Dragon-Seeker committed:
Gegy: fix incorrect boardwalk cullface direction

Dragon-Seeker committed:
Gegy: fix fruit trees trying to generate with saplings for leaves

Dragon-Seeker committed:
SuperCoder7979: Add overgrown mangroves biome

Also add a check for out of chunk setblocks or gathering of info

Dragon-Seeker committed:
SuperCoder7979: Add osa rainforest biome and debug biome command

Command may not be working

Dragon-Seeker committed:
tterrag1098 - Remove village path obstruction clearing, fix structure biomes

Dragon-Seeker committed:
Gegy - allow mangrove roots to connect upwards to solid blocks /// tterrag1098 - Fix a bunch of issues with path processing

Dragon-Seeker committed:
Gegy - use custom jigsaw piece for home tree branches

Dragon-Seeker committed:
Fix Scuba Goggles

Gegy commented on src/generated/test/data/tropicraft/worldgen/configured_feature/huge_golden_leather_fern.json:
```diff
@@ -0,0 +1,53 @@
+{
```
Are these generated/test files being generated intentionally?

Dragon-Seeker replied:
I think I did such back during debug step of attempting to fix some worldgen issues and tag problems. I'll remove it in the latest commit

Dragon-Seeker committed:
Added Experimental Leave decay and Fixed Depreciated constants calls

This leaf decay only targets the mangrove leaves, palm leaves and kapok leaves as the new class for them is the TropicraftExperimentalLeaveBlock

Dragon-Seeker committed:
Test addition of Jappafied Texturepack by Aerod

Dragon-Seeker committed:
Implement PR #380 within 1.17

Dragon-Seeker committed:
Fix Issue #419

Dragon-Seeker committed:
Changed drops for some entity's to be similar to minecraft mobs and Datagened

Added drops for Tropicraft Skeleton, Failgull, Spider, Both Color Basilisk, and the hummingbird

Dragon-Seeker committed:
Whoops forgot to datagen again (;

Dragon-Seeker committed:
Quick adjustment of the Man O War drops And Basilisk

NotActuallyTerry committed:
Point sponsorship button to Love Tropics

Dragon-Seeker committed:
Remove Temporary Test Folder

Dragon-Seeker committed:
Whoops: Append Music Disc Tag to minecrafts tag

Dragon-Seeker committed:
Gegy: [fix: raised mangrove trees can generate at heightmap]

Dragon-Seeker committed:
Initial Test of reimplementing the portal

The portal seems to be generating and working with some issues like generating right on top of it self and

Dragon-Seeker committed:
Portal now works: Issues with re-implementing portal code

Was implementing certain methods incorrectly which was causing issues and problems with the "portal water" was being found even thought it was just water blocks so....

Cojomax99 commented:
Looks like this branch could use a rebase!

Dragon-Seeker committed:
Use POI data to search for portals and fix issue with instant teleport loop

Still issues with the caching of the previous portals

Dragon-Seeker commented:
> Looks like this branch could use a rebase!

That would require a 1.17 branch to target such, right?

Dragon-Seeker committed:
Removed Old Caching method

Dragon-Seeker committed:
Clean Up and changing logger output

Dragon-Seeker commented:
Thanks github. If you rename a branch it will instead of renaming the branch, it will create a new branch with said branch as a base and rename that one meaning it will then delete the "old" one.

Dragon-Seeker committed:
Merge pull request #1 from Dragon-Seeker/1.17-Forge-Portal (Found where all those portals went too.)

More development will need to made to make the portal creation more clean and later when the patchouli book is back, re implement the commented out code involving spawning the Encyclopedia.

Dragon-Seeker committed:
Test reimplementing adding items to portal chest on tropicraft side

Also Move all command registering into the renamed CommandTropics( OG Name: CommandTropicsTeleport)

NotActuallyTerry committed:
Update Love Tropics section of readme

Dragon-Seeker committed:
Update Version appropriately for newer release

Dragon-Seeker committed:
Revert "Update Version appropriately for newer release"

This reverts commit e7ad49878cc55f342a9084823815cda66c3c78f1.

Dragon-Seeker committed:
Update Version appropriately for newer release: Try #2

Dragon-Seeker committed:
"fix" Byte Depreciation issue

NotActuallyTerry committed:
Enable gradle using JDK16

NotActuallyTerry committed:
Remove JDK8 building

Commit:
match all SwordItem instances for pineapple loot (closes #374)

jaskarth committed:
Refactor koas a bit

Dragon-Seeker committed:
Minor change to 1.17 TODO naming

Dragon-Seeker committed:
Datagen Music Disc tag

Arrg I though I did this already!

Gegy commented:
Closing this for now as this has been merged into the 1.17 & 1.18 branches, thank you so much for the insane amount of work you've put into this PR, it has been such a huge help! 💜
